plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/3dtiles/BuildingTransparency/hooks.ts

```diff
@@ -0,0 +1,48 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { BaseFieldProps } from "../../types";
+
+import { useBuildingTransparency } from "./useBuildingTransparency";
+
+type OptionsState = Omit<BaseFieldProps<"buildingTransparency">["value"], "id" | "group" | "type">;
+
+const useHooks = ({
+  value,
+  dataID,
+  onUpdate,
+}: Pick<BaseFieldProps<"buildingTransparency">, "value" | "dataID" | "onUpdate">) => {
+  const [options, setOptions] = useState<OptionsState>({
+    transparency: value.transparency,
+  });
+
+  const handleUpdate = useCallback(
+    <P extends keyof OptionsState>(prop: P, v?: OptionsState[P]) => {
+      setOptions(o => {
+        const next = { ...o, [prop]: v };
+        onUpdate({ id: value.id, type: value.type, group: value.group, ...next });
+        return next;
+      });
+    },
+    [onUpdate, value],
+  );
+
+  const handleUpdateNumber = useCallback(
+    (prop: keyof OptionsState) => (value: number) => {
+      handleUpdate(prop, value);
+    },
+    [handleUpdate],
+  );
+
+  useEffect(() => {
+    setOptions({ ...value });
+  }, [value, onUpdate]);
```

Comment:
Same as the other PR: won't this make a double call to setOptions (once in the handleUpdate, and then another time here)?

Reply:
I fixed 
[fix: prevent unnecessary render](https://github.com/eukarya-inc/reearth-plateauview/pull/122/commits/e67e9db448140504e8532fb23f92e1458457b578)